server/issue.go

```diff
@@ -495,6 +492,51 @@ func httpAPIAttachCommentToIssue(ji Instance, w http.ResponseWriter, r *http.Req
 			errors.WithMessage(err, "failed to attach the comment, postId: "+attach.PostId)
 	}
 
+	if len(post.FileIds) > 0 {
+		go func() {
+			updated := false
+
+			// Upload the files
+			for _, fileId := range post.FileIds {
+				info, ae := api.GetFileInfo(fileId)
+				if ae != nil {
+					continue
+				}
+				// TODO: large file support? Ignoring errors for now is good enough...
+				byteData, ae := api.ReadFile(info.Path)
+				if ae != nil {
+					notifyOnFailedAttachment(ji, ae.Error(), mattermostUserId, info.Path, attach.IssueKey)
+					continue
+				}
+
+				attachments, _, err2 := jiraClient.Issue.PostAttachment(attach.IssueKey, bytes.NewReader(byteData), info.Name)
+
+				if err2 != nil {
+					notifyOnFailedAttachment(ji, err2.Error(), mattermostUserId, info.Path, attach.IssueKey)
+					continue
+				}
+				if attachments != nil {
+					// There will only ever be one attachment at a time.
+					attachment := (*attachments)[0]
+					jiraComment.Body += "\n\nAttachment: !" + attachment.Filename + "!"
+					updated = true
+				}
+			}
+
+			// Update the comment
+			if updated {
+				jiraComment.ID = commentAdded.ID
+				_, _, err2 := jiraClient.Issue.UpdateComment(attach.IssueKey, &jiraComment)
+				if err2 != nil {
+					ji.GetPlugin().API.LogError("failed to update comment: "+err2.Error(), "issue", attach.IssueKey)
+
+					// Report the error to the user so they know they have to update the comment on their own.
+					_, _ = ji.GetPlugin().CreateBotDMtoMMUserId(mattermostUserId, "Failed to update comment to issue %s with file attachments. Please notify your system administrator.", attach.IssueKey)
```

Comment:
What would their administrator need to do?  Why would this happen?

Reply:
@aaronrothschild The sysadmin would have to look at the error in the log to see why the upload didn't happen. I thought this was better than sending the error to the user in a DM.